Add support for tail correction in input

Luthaf opened the pull request:
The documentation should be updated too, it is already out of date concerning the pair interaction cutoff.

g-bauer commented:
I can do the update for the docs.

g-bauer commented:
Just wondering, do you think `tail` as keyword is self-explanatory? Maybe we can make this more clear using `tail_corr` or even `tail_correction`?

Luthaf commented:
> Just wondering, do you think tail as keyword is self-explanatory? Maybe we can make this more clear using tail_corr or even tail_correction?

I don't know. I don't really like `tail_corr`, and `tail_correction` seems a bit long and verbose. I'll think about this!

g-bauer commented:
Verbosity can be good when it makes things more clear. But anyways, I think `tail` is good as well. Do you want to add anything? Else, I'd merge.

Luthaf commented:
> Verbosity can be good when it makes things more clear.

Yes =). I changed it to `tail_correction`, if this bother us too much, we can switch back to `tail` before the first release (or even after, if we change the `input.version` value). I also added the doc, so this fix #28.

g-bauer commented on doc/src/input/electrostatic.md:
```diff
@@ -0,0 +1,77 @@
+long-range interactions splitting. The `kmax` parameter give the number of
+vector to use in the reciprocal space (the long-range part of interactions).
+Usually 7-8 is a good value for pure water, for very periodic charges a lower
+value like 5 can do it, and for more heterogeneous system, higher values of
```
... value, like/such as 5 is sufficient. For more heterogeneous systems, ...

g-bauer commented on doc/src/input/electrostatic.md:
```diff
@@ -0,0 +1,77 @@
+
+## Wolf solver
+
+Wolf method is another method for computing electrostatic interactions presented
```
The Wolf summation method ...

g-bauer commented on doc/src/input/electrostatic.md:
```diff
@@ -0,0 +1,77 @@
+expensive computation in reciprocal space from Ewald by a corrective term, and
+can be expressed as a converging sum over the charged pairs in the system.
+
+It is accessible with the `wolf` keyword in the input files:
```
accessible using the ...

g-bauer commented on doc/src/input/electrostatic.md:
````diff
@@ -0,0 +1,77 @@
+wolf = {cutoff = "11 A"}
+```
+
+The only parameter is a `cutoff`, which should be taken bigger than the
````
which - as a rule of thumb - should be bigger larger than the corresponding cutoff from Ewald summation.

g-bauer commented on doc/src/input/electrostatic.md:
````diff
@@ -0,0 +1,77 @@
+```
+
+The only parameter is a `cutoff`, which should be taken bigger than the
+corresponding Ewald cutoff. A value of 10 A is a good one in pure water.
````
For example, `cutoff = "11 A"` is suitable for pure water.

g-bauer commented on doc/src/input/interactions.md:
```diff
@@ -11,12 +11,14 @@ parameters to describe energies and forces of a molecule is also often called
   - `coulomb` and `charges` describe long-range contributions due to
-  electrostatic interactions.
+  electrostatic interactions;
+  - the `global` section  describe additional parameter that apply to all the
```
section describes additional

g-bauer commented on doc/src/input/pairs.md:
```diff
@@ -0,0 +1,166 @@
+When computing the energy and forces for non-bonded pair interactions, Lumol
+uses a cutoff radius $rc$. This means that the force and energy associated with
+any pair at a distance bigger than $rc$ will be zero. We can use two different
+cutoff scheme, presented in the following section.
```
schemes, ...

g-bauer commented on doc/src/input/pairs.md:
````diff
@@ -0,0 +1,166 @@
+section, and then overridden for specific interactions:
+
+```toml
+# Use a simple cutoff with a radius of 10 A for all the pairs interactions
````
for all pair interactions

g-bauer commented on doc/src/input/pairs.md:
```diff
@@ -0,0 +1,166 @@
+
+### Simple cutoff
+
+This scheme just set the potential $U(r)$ to zero for any distance bigger than
```
sets

g-bauer commented on doc/src/input/pairs.md:
```diff
@@ -0,0 +1,166 @@
+    0 & r > rc
+\end{cases}$$
+
+In the input, this uses a single string containing the cutoff radius as the
```
A suggestion: To use this simple potential truncation, we simply specify a string containing....

g-bauer commented on doc/src/input/pairs.md:
````diff
@@ -0,0 +1,166 @@
+cutoff = {shifted = "8 A"}
+```
+
+### Simple cutoff
````
### Simple cutoff (potential truncation)

g-bauer commented on doc/src/input/pairs.md:
````diff
@@ -0,0 +1,166 @@
+cutoff = "8 A"
+```
+
+### Shifted energy
````
Maybe: Truncation with energy shift

g-bauer commented on doc/src/input/pairs.md:
```diff
@@ -0,0 +1,166 @@
+### Tail correction
+
+Tail corrections (also called long range corrections) are a way to add back the
+missing energy from using a cutoff distance. If the simulated system is
```
Suggestion: ...are a way to account for the error we introduce by cutting off the potential.

g-bauer commented on doc/src/input/pairs.md:
```diff
@@ -0,0 +1,166 @@
+
+Tail corrections (also called long range corrections) are a way to add back the
+missing energy from using a cutoff distance. If the simulated system is
+homogeneous and isotropic after the cutoff (if the pair distribution function
```
beyond the ...

g-bauer commented on doc/src/input/pairs.md:
````diff
@@ -0,0 +1,166 @@
+lj = {x0 = "3 A", k = "5.9 kJ/mol/A^2"}
+tail_correction = false
+```
+
````
Note that using a long range correction will also add an equivalent contribution to the pressure. For details, see Frenkel and Smit.

Luthaf replied:
> Note that using a long range correction will also add an equivalent contribution to the pressure. For details, see Frenkel and Smit.

It does add it, I'll specify this in the doc =)

g-bauer commented on doc/src/input/pairs.md:
```diff
@@ -0,0 +1,166 @@
+## Pairs restrictions
+
+Some force fields define additional restrictions concerning which particles
+should interact together. For example, sometimes bonded particles should not
```
interact with each other?

g-bauer commented on doc/src/input/pairs.md:
````diff
@@ -0,0 +1,166 @@
+```
+
+The possible values for `restriction` are:
+- `"intramolecular"` or `"intra-molecular"` to restrict the potential to only
````
to only affect/act on particles that are

g-bauer commented on doc/src/input/pairs.md:
```diff
@@ -0,0 +1,166 @@
+The possible values for `restriction` are:
+- `"intramolecular"` or `"intra-molecular"` to restrict the potential to only
+  particles in the same molecule;
+- `"intermolecular"` or `"inter-molecular"` to restrict the potential to only
```
to only affect/act on particles that are ...

g-bauer commented on doc/src/input/pairs.md:
```diff
@@ -0,0 +1,166 @@
+- `"intermolecular"` or `"inter-molecular"` to restrict the potential to only
+  particles NOT in the same molecule;
+- `"exclude12"` to exclude particles directly bonded together;
+- `"exclude13"` to exclude particles directly bonded together or forming an
```
maybe use "and" instead of "or" to make clear that `"exclude13"` also contains `"exclude12"`?

g-bauer commented on doc/src/input/pairs.md:
```diff
@@ -0,0 +1,166 @@
+- `"exclude13"` to exclude particles directly bonded together or forming an
+  angle;
+- `"exclude14"` to exclude particles directly bonded together; forming an angle
+  or a dihedral angle;
```
same as above, maybe "and" instead of "or"?

g-bauer commented on doc/src/input/pairs.md:
```diff
@@ -0,0 +1,166 @@
+- `{scale14 = <scaling>}` to exclude particles directly bonded together or
+   forming an angle and to scale interaction between particles at 3 bonds of
+   distance by the given `scaling` factor. The factor must be between 0 and 1.
+
```
I find this difficult to understand. Maybe we can use the nice explanation that you gave above. A suggestion:

... works like `"exclude13"`, i.e. intra molecular interactions between three neighboring particles (connected by  two bonds) will not be computed. Additionally, interactions between the first and the forth (hence the `14` in `scale14`) particle will be computed, but using *scaled* energies and forces. This simply means that the energies and forces are multiplied (linear scaling) by the given `scaling` factor, which must be between 0 and 1.

g-bauer commented on doc/src/input/pairs.md:
```diff
@@ -0,0 +1,166 @@
+   forming an angle and to scale interaction between particles at 3 bonds of
+   distance by the given `scaling` factor. The factor must be between 0 and 1.
+
+## Potentials computations
```
Potential computation (?)

g-bauer commented on doc/src/input/pairs.md:
```diff
@@ -0,0 +1,166 @@
+## Potentials computations
+
+The same potential function (Lennard-Jones, Harmonic, *etc.*) can be computed
+with different method: directly, by sifting it at the cutoff distance, using a
```
methods: directly, by shifting at the cutoff ...

g-bauer commented on doc/src/input/pairs.md:
```diff
@@ -0,0 +1,166 @@
+with different method: directly, by sifting it at the cutoff distance, using a
+table interpolation, *etc.* This is the purpose of computation. The default way
+is to use the mathematical function corresponding to a potential to compute it.
+To use another computation, the `computation` keyword can be used in the
```
To use a different type of computation, ...

Luthaf commented on doc/src/input/electrostatic.md:
```diff
@@ -0,0 +1,77 @@
+The `cutoff` parameter specify the cutoff distance for the short-range and
+long-range interactions splitting. The `kmax` parameter give the number of
+vector to use in the reciprocal space (the long-range part of interactions).
+Usually 7-8 is a good value for pure water, for very periodic charges a lower
```
I changed it to "*for a very periodic charges distribution (like a crystal)*", is it better?

g-bauer commented on doc/src/input/pairs.md:
```diff
@@ -126,27 +130,31 @@ restriction = "intermolecular"
 
 The same potential function (Lennard-Jones, Harmonic, *etc.*) can be computed
-with different method: directly, by sifting it at the cutoff distance, using a
+with different method: directly, by sifting at the cutoff distance, using a
```
with different method**s**: directly, by s**h**ifting ...
